projects/storefrontlib/src/cms-components/myaccount/order/order-details/order-detail-shipping/order-detail-shipping.component.ts

```diff
@@ -1,17 +1,16 @@
-import { Component, OnInit } from '@angular/core';
+import { Component } from '@angular/core';
+import { Order, StateUtils } from '@spartacus/core';
 import { Observable } from 'rxjs';
 import { OrderDetailsService } from '../order-details.service';
 
 @Component({
   selector: 'cx-order-details-shipping',
   templateUrl: './order-detail-shipping.component.html',
 })
-export class OrderDetailShippingComponent implements OnInit {
+export class OrderDetailShippingComponent {
   constructor(protected orderDetailsService: OrderDetailsService) {}
 
-  order$: Observable<any>;
-
-  ngOnInit() {
-    this.order$ = this.orderDetailsService.getOrderDetails();
-  }
+  state$: Observable<
+    StateUtils.LoaderState<Order>
+  > = this.orderDetailsService.getOrderDetailsState();
```

Comment:
Why in some places we map order to null if it's not present, in others we create 2 stream, order and error and in here we have a single stream?

Reply:
The reason is that in reducer by default it has value as `{}`. In two components we are mapping it to null because we don't want to show them at all (if we leave it without mapping {} = true so the template will render). But in another two components, we are using the state as we want to show something else if there is a wrong order code and wrong response. We need to recognize state errors which will help us to show proper content.